### PR TITLE
Expose status reports via notifyAppReady()

### PR DESCRIPTION
**Pull request by nevir**
Handy for tracking whether a deployment succeeded/failed!  We're using this to emit analytics (Mixpanel) events on subsequent launches (success or rollback)

---

Note that I removed the `resumeListener` references, as it was completely unused; lemme know if you want me to preserve it instead!  Ditto the whitespace changes

### CODE REVIEW

**Comment by scottbommarito** [comment]
Looks pretty good to me! Thanks for these changes!

**Comment by lostintangent** [comment]
This looks great to me as well. We really appreciate all the recent contributions @nevir!

**Comment by nevir** [comment]
👍

File: CodePush.js [hunk 1]
```diff
@@ -172,50 +172,41 @@ const notifyApplicationReady = (() => {
...
-      NativeCodePush.recordStatusReported(statusReport);
-      resumeListener && AppState.removeEventListener("change", resumeListener);
-    } catch (e) {
-      log(`Report status failed: ${JSON.stringify(statusReport)}`);
-      NativeCodePush.saveStatusReportForRetry(statusReport);
-      // Try again when the app resumes
-      if (!resumeListener) {
-        resumeListener = (newState) => {
-          newState === "active" && tryReportStatus(resumeListener);
-        };
```
Comment (geof90) [inline CodePush.js]: @nevir This `resumeListener` is our mechanism to retry status reports when the app resumes if the status report call to the server fails. Could we put it back?

Comment (nevir) [inline CodePush.js]: Oho, yup!  Totally missed this bit of recursion

Comment (nevir) [inline CodePush.js]: Alright, it's back via 04e84b5 - PTAL!

Comment (nevir) [inline CodePush.js]: Would it be better if I _don't_ refresh the status (and just let the original report propagate through the status changes?)

Comment (geof90) [inline CodePush.js]: No, I think refreshing it better mimics the original behaviour. Also, between the time it fails and the next time the app resumes, some other process might have already flushed the queue and reported it. So, what you have now is fine. :)

Comment (nevir) [inline CodePush.js]: :+

File: typings/react-native-code-push.d.ts [hunk 2]
```diff
@@ -146,77 +146,99 @@ export interface SyncOptions {
...
+
+    /**
+     * Deployment key used when deploying the previous package.
+     */
+    previousDeploymentKey?: string;
+
+    /**
+     * The label (v#) of the package that was upgraded from.
+     */
+    previousLabelOrAppVersion?: string;
```
Comment (geof90) [inline typings/react-native-code-push.d.ts]: There is also an `appVersion` key in the report, we also report binary updates. For more info: https://github.com/Microsoft/react-native-code-push/blob/master/ios/CodePush/CodePushTelemetryManager.m#L29

Comment (geof90) [inline typings/react-native-code-push.d.ts]: Also, the `package` key is optional

**Comment by geof90** [comment]
It would be great if you could make the `.d.ts` changes as well, but as it is now it looks good to me!

**Comment by nevir** [comment]
Just pushed the `.d.ts` tweaks

**Comment by geof90** [comment]
Thanks!

**Comment by lostintangent** [comment]
@nevir This has now been shipped to NPM via the `1.13.0-beta` release of this plugin. Thanks again for your contribution!

**Comment by nevir** [comment]
sweet :D

**Pull request merged**